src/rgw/rgw_common.h

```diff
@@ -1584,7 +1584,12 @@ class rgw_obj {
       return;
     }
 
-    init_ns(b, key.name.substr(pos + 1), key.name.substr(1, pos -1));
+    // skip setting ns if object itself ends with an underscore
```

Comment:
Don't we do the _temp_ -> __temp_ only at set_obj.. which is called by this function (ie. init/init_ns)? Or are any other places where we do the underscore check for object names before this point.